frontend/windows/handler.go

```diff
@@ -1,24 +1,30 @@
 package windows
 
 import (
+	"context"
+
 	"github.com/Azure/dalec/frontend"
-	"github.com/moby/buildkit/frontend/subrequests/targets"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	bktargets "github.com/moby/buildkit/frontend/subrequests/targets"
 )
 
 const (
-	targetKey = "windowscross"
-	outputKey = "windows"
+	DefaultTargetKey = "windowscross"
```

Comment:
*note: no action needed at this time*

I am thinking ahead to when (or if?) we change the route to windows/cross/container (or possibly windows/container/cross). That is going to require some additional work in the `targets` part of the spec, because we will have one "dalec.target" for both cross-compiled and non-cross-compiled windows containers (both would have `windows` as the top-level path component).

There might be different requirements, like different build dependencies. What do we do about that?

Reply:
could we specify dependencies specifically for `windows/cross` in the targets portion of the spec?

Reply:
That would be difficult, because the client's `opts` is updated to have `dalec.target` set to the first path component. The rest of the target will have been gradually chomped away by the recursive calls to `mux.Handle`.

Reply:
Yeah, I see what you're saying. Trying to think about how this could be dealt with. One possible option is, in addition to `dalec.target`, we could keep an option that is updated with the current handler path. So, the handler registered at `windows/cross/container` would know the path that was taken to get to it. Then, when fetching dependencies for a target, we could first try the full handler path before falling back to `dalec.target`. Just an idea. It does kind of break the nested handler abstraction, though